syscalls/gen-syscall-list/build.rs

```diff
@@ -29,14 +32,29 @@ fn main() {
     let mut text = vec![];
     file.read_to_end(&mut text).unwrap();
     let text = str::from_utf8(&text).unwrap();
-    let file = match File::create(&syscalls_txt_path) {
+    let txt_file = match File::create(&syscalls_txt_path) {
         Ok(x) => x,
         Err(err) => panic!("Failed to create {}: {}", syscalls_txt_path.display(), err),
     };
-    let mut out = BufWriter::new(file);
-    let sysc_re = Regex::new(r#"register_syscall_by_name\([[:space:]]*b"([^"]+)","#).unwrap();
+    let mut txt_out = BufWriter::new(txt_file);
+    let rs_file = match File::create(&build_sbf_syscalls_path) {
+        Ok(x) => x,
+        Err(err) => panic!("Failed to create {}: {}", syscalls_txt_path.display(), err),
+    };
+    let mut rs_out = BufWriter::new(rs_file);
+    writeln!(rs_out, "pub(crate) const SYSCALLS: &[&str] = &[").unwrap();
+    let sysc_re = Regex::new(r#"register_function\([[:space:]]*"([^"]+)","#).unwrap();
     for caps in sysc_re.captures_iter(text) {
         let name = caps[1].to_string();
-        writeln!(out, "{name}").unwrap();
+        writeln!(txt_out, "{name}").unwrap();
+        writeln!(rs_out, "    \"{name}\",").unwrap();
+    }
+    let feature_gate_syscall_re =
+        Regex::new(r#"register_feature_gated_function!\([^"]+"([^"]+)","#).unwrap();
```

Comment:
These two regex matchers depend on the exact format of the `src/lib.rs` file.
Can we either error out if we the number of syscalls changes (triggering someone to modify it build.rs and perhaps notice that something is not right) or add a comment in `src/lib.rs` about its secondary usage?